### DISCUSSION TITLE
Note Recalling Not Working in any browser

### DISCUSSION THREAD

**Discussion by connradolisboa** [coddingtonbear/obsidian-web#196] [Bug Report]
### Describe the bug

I've checked the permissions and they're all enabled, checked the url: frontmatter and it's alright.

Still, the extension doesn't seems to indetify there's already a note with that url. 

### To Reproduce

1. Go to  website
2. Click on obsidian web extension
3. Save page/highlight

### Operating System

Windows

### Browser

Chrome, Arc, Edge, Vivaldi

### Browser Version

The latest in all

**Comment by coddingtonbear** [comment]
It's impossible to say without knowing more about your note templates and notes, but it's extremely likely that the notes you're creating aren't using the necessary frontmatter fields.  In order for the "Note Recall" feature to work, the URL you are visiting must exactly match a pattern as described here: https://github.com/coddingtonbear/obsidian-web/wiki/Page-Notes .

There are of course other things that can be going wrong, too, but I don't have remotely enough information to even make a guess without you submitting a bug report export from the bug-shaped button in the upper-right corner of the extension options page.

**Reply by connradolisboa** [reply]
I've tried using * at the end too and didn't work :( 
Here's the bug report
[obsidian-local-rest-api.bug_report (1).json](https://github.com/user-attachments/files/15635956/obsidian-local-rest-api.bug_report.1.json)

**Comment by coddingtonbear** [comment]
I've had a look at your bug report, and it does look like an error is occurring:

```
        "response": {
          "status": 400,
          "json": {
            "message": "The query you provided could not be processed.\nCannot read properties of null (reading 'toString')",
            "errorCode": 40070
          },
          "text": "{\n  \"message\": \"The query you provided could not be processed.\\nCannot read properties of null (reading 'toString')\",\n  \"errorCode\": 40070\n}"
        }
 
```

I've never seen such an error before from Obsidian, but that's clearly going to prevent matches from being returned.  My *guess* is that Obsidian Local REST API is making some assumptions about the shape of note objects in Obsidian that are being violated by at least one of your notes.  Unfortunately, the error message returned to Obsidian Web doesn't provide any insight into what particular file it's running into trouble with, but there is probably a much-more-helpful error message in Obsidian if you look at its inspector (I believe you can open it with `<ctrl>+<shift>+<i>`).

**Comment by coddingtonbear** [comment]
OK, @connradolisboa -- I've released a new version of Obsidian Local REST API that includes a speculative fix for the problem you're encountering.  Unfortunately, given that I can't reproduce this problem in my vault, I can't really be sure that it will solve your problem, but do give the latest version (2.4.3) a shot and let me know if note recall is now working.

Note that you'll definitely need to tell Obsidian to update the plugin to the latest version before you'll see any changes!

**Reply by connradolisboa** [reply]
Works perfectly man! All the features🫡
